frb_codegen/src/generator/dart/func.rs

```diff
@@ -178,6 +184,81 @@ pub(crate) fn generate_api_func(
     }
 }
 
+pub(crate) fn get_api2wire_prefix(
+    api2wire_func: &str,
+    shared_dart_api2wire_funcs: &Option<Acc<String>>,
+    ir_file: &IrFile,
+    ir_type: &IrType,
+    for_dart_basic_file: bool,
+) -> String {
+    if is_multi_blocks_case(None) && !ir_file.shared {
+        // NOTE: For multi-blocks case, `COMMON_API2WIRE` should have been fetched by
+        // `DartApiSpec::from()` according to the whole frb generation routine.
+        FETCHED_FOR_COMMON_API2WIRE.with(|data| {
+            assert!(*data.borrow_mut(), "COMMON_API2WIRE not fetched before");
```

Comment:
by avoiding global variables, we can automatically avoid this kind of implicit dependency on another far-away code, then our code gets easier to debug/understand/maintain